packages/material-ui/src/Snackbar/Snackbar.d.ts

```diff
@@ -13,12 +13,12 @@ export interface SnackbarProps
       React.HTMLAttributes<HTMLDivElement> & Partial<TransitionHandlerProps>,
       SnackbarClassKey
     > {
-  action?: React.ReactElement<any> | React.ReactElement<any>[];
+  action?: React.ReactNode;
```

Comment:
Since this property is passed to `SnackBarContent` it should explicitly reference that type like `SnackBarContentProps['action']`. Same for `message`.